### PR TITLE
mDeleteXX on security components

### LINKED ISSUE #619: Delete by query Users/Roles/Profiles
Hello,

Because we can't perform a `.deleteDocument` on the index `%kuzzle` we can't delete users/roles/profiles with a query (or simply with an array of IDs).

We need to have a way to delete mutli user/profile/role in Kuzzle and in SDKs.

### PR DESCRIPTION
**Pull request by benoitvidis**
Fixes #619

This PR exposes the following methods:

* `mDeleteProfile`
* `mDeleteRole`
* `mDeleteUser`

in the `security` controller.

# Boyscouting - breaking change

In order to align actions naming between #616 and because the `m` prefix already specifies the plurality, `security:mGetProfiles` and `security:mGetRoles` are respectively renamed to `mGetProfile` and `mGetRole`

### CODE REVIEW

**Comment by benoitvidis** [comment]
Just realized http routes were missing... sorry -_-;

File: lib/config/httpRoutes.js [hunk 1]
```diff
@@ -128,6 +128,9 @@ module.exports = [
   {verb: 'post', url: '/users/:_id/_createRestricted', controller: 'security', action: 'createRestrictedUser'},
   {verb: 'post', url: '/users/_create', controller: 'security', action: 'createUser'},
   {verb: 'post', url: '/users/:_id/_create', controller: 'security', action: 'createUser'},
+  {verb: 'post', url: '/profiles/_mdelete', controller: 'security', action: 'mDeleteProfile'},
```
Comment (benoitvidis) [inline lib/config/httpRoutes.js]: The http RFC is not clear about allowing a body in a DELETE request. To be safe, _mdelete calls are set as POST
http://stackoverflow.com/questions/299628/is-an-entity-body-allowed-for-an-http-delete-request

File: features/step_definitions/profiles.js [hunk 2]
```diff
@@ -247,7 +247,7 @@ var apiSteps = function () {
 
     main = function (callbackAsync) {
       setTimeout(() => {
-        this.api.mGetProfiles(body)
+        this.api.mGetProfile(body)
```
Comment (benoitvidis) [inline features/step_definitions/profiles.js]: boycouting - align action names with #616

File: lib/api/controllers/securityController.js [hunk 3]
```diff
@@ -550,3 +564,57 @@ function resetProfiles () {
...
+ * @param {Request} request
+ * @returns {Promise.<*>}
+ */
+function mDelete (kuzzle, type, request) {
+  const
+    action = _.upperFirst(type);
+
+  assertHasBody(request, `security:${action}`);
+  assertBodyHasAttribute(request, 'ids', `security:${action}`);
+  if (!Array.isArray(request.input.body.ids)) {
```
Comment (ballinette) [inline lib/api/controllers/securityController.js]: nitpicking: replace it by `assertHasIndexAndCollection` introduced here: https://github.com/kuzzleio/kuzzle/pull/616/files/eb4baa72454df15e01cc11bdc246029bbd5e52e2..98450af8137849d5e3b0942b949ab64e248c765b#diff-91a156dc5237febe996201236bc18f66R44

(implies a dependency with #616)

Comment (dbengsch) [inline lib/api/controllers/securityController.js]: read `AssertBodyAttributeType`

File: lib/api/controllers/securityController.js [hunk 4]
```diff
@@ -550,3 +564,57 @@ function resetProfiles () {
...
+ * @param {Kuzzle} kuzzle
+ * @param {string.<profile|role|user>} type
+ * @param {Request} request
+ * @returns {Promise.<*>}
+ */
+function mDelete (kuzzle, type, request) {
+  const
+    action = _.upperFirst(type);
+
+  assertHasBody(request, `security:${action}`);
```
Comment (dbengsch) [inline lib/api/controllers/securityController.js]: shouldn't it be `security:delete${action}` ? (if depends on #616 as suggested below, it becomes useless)

File: lib/config/httpRoutes.js [hunk 5]
```diff
@@ -128,8 +128,11 @@ module.exports = [
   {verb: 'post', url: '/users/:_id/_createRestricted', controller: 'security', action: 'createRestrictedUser'},
   {verb: 'post', url: '/users/_create', controller: 'security', action: 'createUser'},
   {verb: 'post', url: '/users/:_id/_create', controller: 'security', action: 'createUser'},
-  {verb: 'post', url: '/profiles/_mget', controller: 'security', action: 'mGetProfiles'},
-  {verb: 'post', url: '/roles/_mget', controller: 'security', action: 'mGetRoles'},
+  {verb: 'post', url: '/profiles/_mdelete', controller: 'security', action: 'mDeleteProfile'},
```
Comment (dbengsch) [inline lib/config/httpRoutes.js]: I just aligned all `m` action in camel case in #616 ...

**Comment by xbill82** [comment]
It makes me sad, being a very effective tool to express plurality available in the english language (kind of adding an "s" at the end of a word), that we simply do the other way around (putting an "m" at the beginning of a singular word).

Would `getUsers`, `getRoles` and `getProfiles` be so ugly?

**Pull request merged**